src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node first;
+    private Node last;
+
+    private int size;
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> next;
+        private Node<T> previous;
+
+        public Node(T element) {
+            this.element = element;
+        }
+    }
+
+    void isIndexOk(int index, int size) {
+        if (index < 0 || index + 1 > size) {
+            throw new IndexOutOfBoundsException("Index can`t be < 0 and > size, index = " + index
+                    + " size = " + size);
+        }
+    }
+
+    public Node<T> getNode(int index) {
```

Comment:
Methods used only in a class must be private

All private methods should be below public methods
```suggestion
    private Node<T> getNode(int index) {
```